server/src/main/java/org/eclipse/lsp/cobol/service/CobolTextDocumentService.java

```diff
@@ -256,6 +260,17 @@ public void observerCallback(@NonNull RunAnalysisEvent event) {
     docs.forEach((key, value) -> analyzeDocumentFirstTime(key, value.getText(), event.verbose));
   }
 
+  @Override
+  @CheckServerShutdownState
+  public CompletableFuture<AnalysisResult> analysis(@NonNull JsonObject json) {
+    RetrieveAnalysisEvent event = new Gson().fromJson(json.toString(), RetrieveAnalysisEvent.class);
+    return CompletableFuture.supplyAsync(
+        () -> Optional.ofNullable(docs.get(event.getUri())).map(CobolDocumentModel::getAnalysisResult).orElse(null),
+        executors.getThreadPoolExecutor())
+        .whenComplete(
+            reportExceptionIfThrown(createDescriptiveErrorMessage("range resolving", event.getUri())));
```

Comment:
I think that you need to change to something like "analysis retrieving".

Reply:
fixed